atomics/include/desul/atomics/Thread_Fence_HIP.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+
+#ifndef DESUL_ATOMICS_THREAD_FENCE_HIP_HPP_
+#define DESUL_ATOMICS_THREAD_FENCE_HIP_HPP_
+
+#include <desul/atomics/Common.hpp>
+
+namespace desul {
+namespace Impl {
+
+// clang-format off
+inline __device__ void device_atomic_thread_fence(MemoryOrderRelease, MemoryScopeDevice) { __threadfence();        }
```

Comment:
Since this and the CUDA one are character for character identical, is it worth keeping them separate?

Reply:
I tend to agree that it would be nice to revisit but it would raise more questions because we could have similar considerations regarding `Fetch_Op_{HIP,CUDA}.hpp` and even possibly `Compare_Exchange_{HIP,CUDA}.hpp`.
Therefore I'd rather keep them as is and revisit later as we judge fit.

Reply:
Works for me.